### PR TITLE
Provide `equals` and `hashCode` on `IterableWrapper` again (fixing 2.13.9 regression)

### LINKED ISSUE #12683: IterableWrapper no longer delegates equals
## Reproduction steps

```scala
scala 2.13.8> import scala.jdk.CollectionConverters._
import scala.jdk.CollectionConverters._

scala 2.13.8> Nil.asJavaCollection == Nil.asJavaCollection
val res0: Boolean = true

scala 2.13.8> Set("foo").asJavaCollection == Set("foo").asJavaCollection
val res1: Boolean = true
```

```scala
scala 2.13.10> import scala.jdk.CollectionConverters._
import scala.jdk.CollectionConverters._

scala 2.13.10> Nil.asJavaCollection == Nil.asJavaCollection
val res0: Boolean = false

scala 2.13.10> Set("foo").asJavaCollection == Set("foo").asJavaCollection
val res1: Boolean = false
```

## Problem

It would be nice if it behaved the same. A wrapper should delegate if possible.

Arguably, there is no contract for structural equality:
```
scala> def f = new java.util.AbstractCollection[Int] {
     | def size = 3; def iterator = Iterator(1,2,3).asJava }
def f: java.util.AbstractCollection[Int]

scala> f == f
val res0: Boolean = false
```
Therefore, there should be no expectation here.

Usage of correct API restores expectations:
```
scala 2.13.10> Set("foo").asJava == Set("foo").asJava
val res2: Boolean = true
```

Reported at https://github.com/scala/scala/pull/10104#issuecomment-1297233492

### PR DESCRIPTION
**Pull request by som-snytt**
Followup to #10104 

Fixes scala/bug#12683 (regression in 2.13.9)

### CODE REVIEW

File: src/library/scala/collection/convert/JavaCollectionWrappers.scala [hunk 1]
```diff
@@ -56,7 +57,15 @@ private[collection] object JavaCollectionWrappers extends Serializable {
   }
 
   @SerialVersionUID(3L)
-  class IterableWrapper[A](val underlying: Iterable[A]) extends ju.AbstractCollection[A] with IterableWrapperTrait[A] with Serializable
+  class IterableWrapper[A](val underlying: Iterable[A]) extends ju.AbstractCollection[A] with IterableWrapperTrait[A] with Serializable {
+    import scala.runtime.Statics._
+    override def equals(other: Any): Boolean =
+      other match {
+        case other: IterableWrapper[A @uc] => underlying.equals(other.underlying)
```
Comment (sjrd) [inline src/library/scala/collection/convert/JavaCollectionWrappers.scala]: That's not a safe `@unchecked`. Consider making `A` covariant and then match on `IterableWrapper[Any]`.

Comment (som-snytt) [reply src/library/scala/collection/convert/JavaCollectionWrappers.scala]: Thanks, cut/paste. I considered just `[_]`.

**Comment by som-snytt** [comment]
Per the test, Iterator does not delegate, so even if the wrapped iterators are `eq` etc. Not obvious to me that there is any useful semantics. Alternatively, obviously equal if `eq` or wrapped iterators are `eq`.

**Review by lrytz — APPROVED** [review]
Thank you!

**Pull request merged**